File: 0-easy/json_menu_ids/json_menu_ids.php
<?php
$fileName = $argv[1];
$lines = file($fileName);

foreach ($lines as $line) {
    $line = trim($line);
    if (!trim($line)) {
        continue;
    }
    $result = json_decode($line);
    $sum = 0;
    foreach ($result->menu->items as $item) {
        if (isset($item->label)) {
            $sum += intval($item->id);
        }
    }
    echo $sum."\n";
}

File: 0-easy/json_menu_ids/json_menu_ids.rb
require 'json'

File.open(ARGV[0]).each_line do |line|
  line = line.strip
  next if line.empty?
  json =  JSON.parse(line)
  sum = 0;
  json["menu"]["items"].each do |items|
    if (items && !items['label'].nil?)
      sum += items['id'].to_i
    end
  end
  puts sum
end
